docs/data/joy/customization/dark-mode/DarkModeByDefault.js

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import { CssVarsProvider } from '@mui/joy/styles';
+import Sheet from '@mui/joy/Sheet';
+import Chip from '@mui/joy/Chip';
+import Typography from '@mui/joy/Typography';
+
+const useEnhancedEffect =
+  typeof window !== 'undefined' ? React.useLayoutEffect : React.useEffect;
```

Comment:
Would it be simpler?

```suggestion
import { unstable_useEnhancedEffect as useEnhancedEffect } from '@mui/utils';

```

I assume that we will want to make this module stable, since we use it everywhere.

Reply:
> Would it be simpler?

I don't think so. The first time I see this import I have no idea what it does so I have to look into the source code anyway. It is even worse for developers who want to copy the code or open the CodeSandbox.

Reply:
Oh, interesting, agree that there is value in inlining to be clearer about what happens.